modules/ensemble/lib/action/get_network_info_action.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/device.dart';
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/framework/scope.dart';
+import 'package:ensemble/framework/stub/location_manager.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:get_it/get_it.dart';
+
+import '../framework/stub/network_info.dart';
+
+class GetNetworkInfoAction extends EnsembleAction {
+  NetworkInfoManager networkInfo = GetIt.I<NetworkInfoManager>();
+  EnsembleAction? onSuccess;
+  EnsembleAction? onError;
+  EnsembleAction? onDenied;
+  EnsembleAction? onLocationDisabled;
+
+  GetNetworkInfoAction({super.initiator,
+    super.inputs,
+    this.onSuccess,
+    this.onError,
+    this.onDenied,
+    this.onLocationDisabled});
+
+  factory GetNetworkInfoAction.from({Invokable? initiator,dynamic payload}) =>
```

Comment:
you don't need this. just the next one.

Reply:
I do need it for javascript calls, when coming from js, it's not yaml it's json